egs/dampvsep/ConvTasNet/train.py

```diff
@@ -0,0 +1,168 @@
+import os
+import argparse
+from pathlib import Path
+import json
+
+import torch
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
+
+from asteroid.models import ConvTasNet
+from asteroid.data import DAMPVSEPDataset
+from asteroid.engine.optimizers import make_optimizer
+from asteroid.engine.system import System
+from asteroid.losses import SingleSrcNegSTOI
+
+
+# Keys which are not in the conf.yml file can be added here.
+# In the hierarchical dictionary created when parsing, the key `key` can be
+# found at dic['main_args'][key]
+# By default train.py will use all available GPUs. The `id` option in run.sh
+# will limit the number of available GPUs for train.py .
+parser = argparse.ArgumentParser()
+parser.add_argument("--exp_dir", default="exp/tmp", help="Full path to save best validation model")
+
+
+def main(conf):
+    # Define dataloader using ORIGINAL mixture.
+    dataset_kwargs = {
+        "root_path": Path(conf["data"]["root_path"]),
+        "task": conf["data"]["task"],
+        "sample_rate": conf["data"]["sample_rate"],
+        "num_workers": conf["training"]["num_workers"],
+        "mixture": conf["data"]["mixture"],
+    }
+
+    train_set = DAMPVSEPDataset(
+        split=f"train_{conf['data']['train_set']}",
+        random_segments=True,
+        segment=conf["data"]["segment"],
+        samples_per_track=conf["data"]["samples_per_track"],
+        **dataset_kwargs,
+    )
+
+    val_set = DAMPVSEPDataset(split="valid", **dataset_kwargs)
+
+    train_loader = DataLoader(
+        train_set,
+        shuffle=True,
+        batch_size=conf["training"]["batch_size"],
+        num_workers=conf["training"]["num_workers"],
+        drop_last=True,
+    )
+
+    val_loader = DataLoader(
+        val_set, shuffle=False, batch_size=1, num_workers=conf["training"]["num_workers"]
+    )
+
+    conf["masknet"].update({"n_src": conf["data"]["n_src"]})
```

Comment:
This is like this in the other recipes because `n_src` is needed in the dataset, and can be changed inside of it. 
If `n_src` is not needed in the data, you can directly add it in conf["masknet"] from the yaml file.